runtime/commonTest/src/kotlinx/serialization/json/JsonUseDefaultOnNullTest.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json
+
+import kotlinx.serialization.*
+import kotlin.test.*
+
+class JsonUseDefaultOnNullTest : JsonTestBase() {
+    @Serializable
+    data class WithBoolean(val b: Boolean = false)
+
+    @Serializable
+    data class WithEnum(val e: SampleEnum = SampleEnum.OptionC)
+
+    @Serializable
+    data class MultipleValues(
+        val data: StringData,
+        val data2: IntData = IntData(0),
+        val i: Int = 42,
+        val e: SampleEnum = SampleEnum.OptionA,
+        val foo: String
+    )
+
+    val json = Json(JsonConfiguration.Default.copy(treatNullAsMissing = true, isLenient = true))
+
+    private inline fun <T> doTest(inputs: List<String>, expected: T, serializer: KSerializer<T>) {
+        for (input in inputs) {
+            parametrizedTest(json) {
+                assertEquals(expected, parse(serializer, input), "Failed on input: $input")
+            }
+        }
+    }
+
+    @Test
+    fun testBoolean() = doTest(
```

Comment:
Please consider giving tests meaningful names for the sake of better readability